### PR TITLE
fixed prevent default

### PR DESCRIPTION
**Pull request by xr0master**
Avoid 'Unable to preventDefault inside passive event listener invocation.' message.
Support compatibility with old browsers

Closes https://github.com/chartjs/chartjs-plugin-annotation/issues/159 https://github.com/chartjs/chartjs-plugin-annotation/pull/122

### CODE REVIEW

**Comment by xr0master** [comment]
@benmccann Used the same logic as Chart.js

**Comment by benmccann** [comment]
@xr0master can you point me to where Chart.js does this logic? I didn't see it

**Comment by xr0master** [comment]
@benmccann https://github.com/chartjs/Chart.js/blob/master/src/platforms/platform.dom.js#L127

**Comment by benmccann** [comment]
Could we call [helpers.addEvent](https://github.com/chartjs/Chart.js/blob/master/src/platforms/platform.dom.js#L457) instead of copying the code into this repo?

**Comment by xr0master** [comment]
Wut? You do it in your code already. This plugin creates the eventHandlers in the dispatcher method and binds it to chart by helpers.addEvent.

File: src/helpers.js [hunk 1]
```diff
@@ -168,6 +168,22 @@ module.exports = function(Chart) {
 			.slice(0, 1)[0]; // return only the top item
 	}
 
+	var doesSupportPassiveEvent = (function() {
```
Comment (benmccann) [inline src/helpers.js]: let's call this `supportsEventListenerOptions` for consistency with the main repo

maybe also add a comment:
```
/**
 * Copied from https://github.com/chartjs/Chart.js/blob/4f722ab619be071e90dd9ea67db27032dc9edd2a/src/platforms/platform.dom.js#L109
 */
```

**Comment by xr0master** [comment]
The issue is that Chart.js uses passive events if possible. Therefore, we check if it is possible too, then we don't call the preventDefault method. You cannot call this method for a passive event, this is the meaning of being passive.

**Comment by benmccann** [comment]
Yeah, sorry, looked too quickly. This looks good to me. Just one minor comment. If you don't mind updating quickly I can merge this

File: src/helpers.js (hunk 1 above)
Comment (xr0master) [reply src/helpers.js]: OK, no problem. Just I like to do question variable for boolean value

**Comment by benmccann** [comment]
@xr0master I noticed there's another pending PR open for this. Do you have thoughts about @KumarNarendra's alternate suggested fix in https://github.com/chartjs/chartjs-plugin-annotation/pull/122#issuecomment-469690185 ?

**Comment by xr0master** [comment]
His suggestion is to reduce performance to fix the useless call of preventDefault method. But how can you do it? The event listeners are in Chart.js code.

**Comment by benmccann** [comment]
What's the reason for calling `e.preventDefault()` in old browsers in this PR vs never doing it like https://github.com/chartjs/chartjs-plugin-annotation/pull/122 is proposing?

**Comment by xr0master** [comment]
God knows why the author of this plugin uses the preventDefault method. Perhaps that was the reason. You need to decide which solution you like more.

PS My opinion is that this plugin should be completely rewritten for new solutions. Maybe when the Chart.js new version 3 comes out.

**Comment by laurynasn** [comment]
Are there any plans to release either this or #160 in the near future?

**Comment by Borzenko** [comment]
@benmccann are you going to release this fix?

**Comment by stockiNail** [comment]
@benmccann I have the same issue.

**Comment by benmccann** [comment]
I'm not really the maintainer for this plugin and it's not clear to me what the right fix is. I've merged a few PRs just to clear out the review queues on some of the plugins, but I didn't author this plugin, don't use it, and am not particularly interested in taking it over. Even if I did merge one of these PRs, I don't have permission to cut new releases. If there's anyone with a history of contributing to Chart.js projects that wants to contribute, please let us know in the #dev channel on Slack

**Comment by stockiNail** [comment]
@benmccann apologize. I didn't want to push you to find out a solution.
I have added a post just to track this PR.
Of course, the expectation is that someone will take care and fix. But I'm also aware that it's not so easy.

It's difficult for me to support the development of this project due to my poor knowledge of javascript.

Nevertheless I have tested the sample setting as comments the llines ([here](https://github.com/chartjs/chartjs-plugin-annotation/blob/7037eea71314613a8af83d2298241841cbc34c4f/src/events.js#L83) and [here](https://github.com/chartjs/chartjs-plugin-annotation/blob/7037eea71314613a8af83d2298241841cbc34c4f/src/events.js#L98)) where `preventDefault()` is called and I can confirm that  it works without any message.

I dont' know if it can help

**Comment by loganknecht** [comment]
Hello! I've looked at the PR and this seems sane. Is there a chance this can be merged? I understand that there isn't really a maintainer for this, but it would be nice to have this merged to fix this bug.

If not, no problem - it is a bit of a strange issue!

**Comment by Ribeiro-Tiago** [comment]
I would imagine this PR also closes https://github.com/chartjs/chartjs-plugin-annotation/issues/179

**Review by kurkle — APPROVED** [review]

**Pull request merged**